src/main/scala/za/co/absa/spark/commons/implicits/StructFieldImplicits.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.implicits
+
+import org.apache.spark.sql.types._
+import scala.util.Try
+
+object StructFieldImplicits {
+  implicit class StructFieldEnhancements(val structField: StructField) {
+    def getMetadataString(key: String): Option[String] = {
```

Comment:
Just realized, this is more `getStringMetadata` as opposed to `getMetadataString` which can be interpreted as `getMetadataAsString`. 

Then again I could be alone in this perception. @benedeki @dk1844 ?

Reply:
I tend to agree with you. But maybe I can create a new method called `getMetadataAsString`

Reply:
The As gives the impression of casting. And that's not happening iirc.
`.getMetadata...` groups these methods with `StructField` in the IDE hinter. And semantically doesn't feel wrong to me. 🤷

Reply:
Yes, casting is not happening. That is what I am saying. But it sounds like it will give you anything there is as a string. 

That is why I would suggest a rename to `getStringMetadata`

Reply:
My perception:
* `getMetadataAsString` suggests casting
* `getStringMetadata` OK
* `getMetadataString` OK the similar way with added benefit of grouping the metadata accessors together

Reply:
I started thinking why is this an implicit on StructFields when this is actually an implicit on metadata. `getOptString` or something

Reply:
Good point. I would actually keep it into StructFioeldsImplicits but the implicit class would be called `StructFieldMetadataEnhancement`, since there is other functionality to be added into StructFieldImplicits in #23 . Thus, the functions can be called `getOptString`, `getOptChar` and `getOptStringAsBoolean`

Reply:
`getOptString` sound better to me. I have noticed that you left the original form in the tests, though.